workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder-solo.md

```diff
@@ -0,0 +1,320 @@
+# From Zero to Binder!
+
+Sarah Gibson, _The Alan Turing Institute_
+
+[**The Turing Way**](https://github.com/alan-turing-institute/the-turing-way) - making reproducible Data Science "too easy not to do"
+
+Based on Tim Head's _Zero-to-Binder_ workshops which can be found here: http://bit.ly/zero-to-binder and http://bit.ly/zero-to-binder-rise
+
+As you work through this tutorial, you will be launching a lot of Binder windows.
+Binder can take a long time to load, particularly if you're working with R, but this doesn't necessarily mean that your Binder will fail to launch.
+You can always refresh the window if you see the "... is taking longer to load, hang tight!" message.
+
+## Running Code is more complicated than Displaying Code
+
+GitHub is a great service for sharing code, but the contents are **static**.
+
+How could you _run_ a GitHub repository **without installing complicated requirements**?
+Or even **in your browser**?
+
+To run code, you need:
+* Hardware on which to run the code
+* Software, including:
+  * The code itself
+  * The programming language (e.g. Python, R)
+  * Relevant packages (e.g. pandas, matplotlib, tidyverse, ggplot)
+
+<a name="binder"></a>
+## What Binder Provides
+
+[Binder](https://mybinder.org) is a service that provides your code and the hardware and software to execute it.
+
+You can create a link to a **live, interactive** version of your code!
+
+* An example binder link:
+
+  > **https://mybinder.org/v2/gh/trekhleb/homemade-machine-learning/master?filepath=notebooks%2Fanomaly_detection%2Fanomaly_detection_gaussian_demo.ipynb**
+
+  From this repo: **https://github.com/trekhleb/homemade-machine-learning**
+
+  * Notice that the Binder link has a similar structure to the GitHub repo link (<github-username>/<github-repo-name>)
+  * The "?filepath=" argument opens a specific notebook (`.ipynb` file) in the repo
+
+## 1. Creating a repo to Binderize
+
+**TO DO:** :vertical_traffic_light:
+
+1) Create a new repo on GitHub called "my-first-binder".
+   * Don't forget to initialise with a README!
+2) Create a file called `hello.py` via the web interface with `print("Hello from Binder!")` on the first line and commit to master
+
+**N.B.:** GitHub repos **must** be _public_ in order to work with Binder.
+
+<html>
+<img src="figures/create_repo.png" alt="create_repo" height="562" width="671">
+<img src="figures/create_hello.png" alt="create_rhello" height="999" width="1006">
```

Comment:
A lot of the images are slightly squashed; could you preserve the aspect ratio somehow? I'm not sure whether full width (`width="100%"`) or specifying only the width in pixels would be better for this particular document.

Reply:
So I halved the pixel dimensions of the original images to stop them from being massive, but I don't know much else about html images unfortunately 🤷‍♀️